src/removeNPMAbsolutePaths.js

```diff
@@ -36,20 +36,26 @@ function processFile(filePath, opts) {
   })
     .then((data) => {
       let writeFile = false;
+      let errorFound = false;
       let obj;
       try {
         obj = JSON.parse(data);
       } catch (err) {
-        throw createError(`Malformed package.json file at "${filePath}"`, err);
+        writeFile = false;
+        errorFound = true;
+        console.log(`Malformed package.json file at "${filePath}"`);
+        console.log('skipping...');
       }
 
-      Object.keys(obj).forEach((key) => {
-        const shouldBeDeleted = opts.fields ? (opts.fields.indexOf(key) !== -1) : (key[0] === '_');
-        if (shouldBeDeleted) {
-          delete obj[key];
-          writeFile = true;
-        }
-      });
+      if(!errorFound) {
```

Comment:
Thus, this is not necessary because the function exit on the error and doesn't get here if it's not parseable.